manufacturers/Enginko/models/EGK/uplink_decoder.js

```diff
@@ -0,0 +1,374 @@
+{
+    const uplinkId = payload.substring(0, 2);
+    switch(uplinkId){
+            case '01':
+                return parseTimeSync(payload);
+
+            case '04':
+                return parseTER(payload);
+
+            default:
+                break;
+    }
+    return null
+}
+
+function ngsildInstance(value, time, unit, dataset_suffix) {
+    var ngsild_instance = {
+        type: 'Property',
+        value: value,
+        observedAt: time
+    }
+    if (unit !== null) {
+        ngsild_instance.unitCode = unit
+    }
+    if (dataset_suffix !== null) {
+        ngsild_instance.datasetId = 'urn:ngsi-ld:Dataset:' + dataset_suffix
+    }
+    return ngsild_instance
+}
+
+function ngsildWrapper(input, time) {
+
+    let ngsild_payload = {};
+    let temperature_data = [];
+    let humidity_data = [];
+    let pressure_data = [];
+    input.forEach(item => {
+        if(item.variable == 'date')
+        {
+            time = item.value;
+        }
+        if(item.variable == 'syncID')
+        {
+            ngsild_payload.syncID = ngsildInstance(item.value, 
+                                                   time, 
+                                                   null, 
+                                                   null);
+        }
+        if(item.variable == 'syncVersion')
+        {
+            ngsild_payload.syncVersion = ngsildInstance(item.value, 
+                                                        time, 
+                                                        null, 
+                                                        null);
+        }
+        if(item.variable == 'applicationType')
+        {
+            ngsild_payload.applicationType = ngsildInstance(item.value, 
+                                                            time, 
+                                                            null, 
+                                                            null);
+        }
+        if(item.variable == 'temperature')
+        {
+            temperature_data.push(ngsildInstance(item.value, 
+                                                 time, 
+                                                 'CEL', 
+                                                 'Raw'));
+        }
+        if(item.variable == 'humidity')
+        {
+            humidity_data.push(ngsildInstance(item.value, 
+                                              time, 
+                                              'P1', 
+                                              'Raw'));
+        }
+        if(item.variable == 'pressure')
+        {
+            pressure_data.push(ngsildInstance(item.value, 
+                                              time, 
+                                              'A97', 
+                                              'Raw'));
+        }
+    });
+
+    if(temperature_data.length != 0)
+    {
+        ngsild_payload.temperature = temperature_data;
+        ngsild_payload.humidity = humidity_data;
```

Comment:
Is it not possible that temperature is not empty while humidity is ?

Reply:
I would say no, but you are right. I am not going take any risks and add the tests.